tabular/src/autogluon/tabular/models/tabpfnmix/_internal/models/foundation/foundation_transformer.py

```diff
@@ -0,0 +1,186 @@
+
+import einops
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from .embedding import FoundationEmbeddingX, FoundationEmbeddingYFloat, FoundationEmbeddingYInteger
+
+
+class FoundationTransformer(nn.Module):
+
+    def __init__(
+            self,
+            n_features: int,
+            n_classes: int, 
+            dim: int,
+            n_layers: int,
+            n_heads: int,
+            attn_dropout: float,
+            y_as_float_embedding: bool,
+            use_pretrained_weights: bool,
+            path_to_weights: str,
+        ) -> None:
+        
+        super().__init__()
+
+        self.n_features = n_features
+        self.n_classes = n_classes
+        self.dim = dim
+        self.n_layers = n_layers
+        self.n_heads = n_heads
+        self.attn_dropout = attn_dropout
+        self.y_as_float_embedding = y_as_float_embedding
+
+        self.x_embedding = FoundationEmbeddingX(dim, n_features)
+
+        if self.y_as_float_embedding:
+            self.y_embedding = FoundationEmbeddingYFloat(dim)
+        else:
+            self.y_embedding = FoundationEmbeddingYInteger(n_classes, dim)
+
+        self.layers = nn.ModuleList([])
+
+        for _ in range(n_layers):
+
+            att = MultiheadAttention(dim, n_heads)
+
+            self.layers.append(nn.ModuleDict({
+                'layer_norm1': nn.LayerNorm(dim),
+                'attention': att,
+                'layer_norm2': nn.LayerNorm(dim),
+                'linear1': nn.Linear(dim, dim*4),
+                'linear2': nn.Linear(dim*4, dim),
+            }))
+
+        self.final_layer1 = nn.Linear(dim, dim*4)
+        self.final_layer2 = nn.Linear(dim*4, n_classes)
+
+        if use_pretrained_weights:
+            self.load_state_dict(torch.load(path_to_weights, weights_only=True))
```

Comment:
`# nosec B614` to exclude it from bandit check

Reply:
removed this code